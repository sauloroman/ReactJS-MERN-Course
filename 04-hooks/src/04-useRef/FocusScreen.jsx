import { useRef } from "react"

export const FocusScreen = () => {

  const inputRef = useRef()

  const onFocusInput = () => {
    inputRef.current.focus()
  }

  return (
    <>
      <h1 className="heading">UseRef &mdash; InputFocus</h1>

      <form action="#" className="form">
        <div className="form__field">
          <input ref={inputRef} placeholder="Presiona el boton..." type="text" className="form__input" />
        </div>

        <button onClick={ onFocusInput } className="btn">Seleccionar el input</button>
      </form>
    </>
  )
}
