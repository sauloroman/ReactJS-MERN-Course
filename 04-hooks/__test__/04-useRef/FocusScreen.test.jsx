import { fireEvent, render, screen } from "@testing-library/react"
import { FocusScreen } from "../../src/04-useRef/FocusScreen"

describe("Tests in FocusScreen Component", () => {

  test("It must match with snapshot", () => {
    const { container } = render(<FocusScreen />)
    expect( container ).toMatchSnapshot()
  })

  test('it must focus the input field when button is pushed', () => {
    render(<FocusScreen />)

    const button = screen.getByText("Seleccionar el input")
    fireEvent.click( button )

    const inputEle = screen.getByRole('textbox')

    expect( inputEle.focus ).toBeTruthy()

  })

})