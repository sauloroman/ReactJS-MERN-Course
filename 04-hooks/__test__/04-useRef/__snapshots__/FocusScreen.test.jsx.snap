// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in FocusScreen Component It must match with snapshot 1`] = `
<div>
  <h1
    class="heading"
  >
    UseRef â€” InputFocus
  </h1>
  <form
    action="#"
    class="form"
  >
    <div
      class="form__field"
    >
      <input
        class="form__input"
        placeholder="Presiona el boton..."
        type="text"
      />
    </div>
    <button
      class="btn"
    >
      Seleccionar el input
    </button>
  </form>
</div>
`;
